Lib/imaplib.py

```diff
@@ -588,6 +648,48 @@ def getquotaroot(self, mailbox):
         return typ, [quotaroot, quota]
 
 
+    def idle(self, duration=None):
+        """Return an Idler: an iterable context manager for the IDLE command
+
+        :param duration:  Maximum duration (in seconds) to keep idling,
+                          or None for no time limit.
+                          To avoid inactivity timeouts on servers that impose
+                          them, callers are advised to keep this <= 29 minutes.
+                          See the note below regarding IMAP4_stream on Windows.
+        :type duration:   int|float|None
+
+        The context manager sends the IDLE command upon entry, produces
+        responses via iteration, and sends DONE upon exit.
+        It represents responses as (type, datum) tuples, rather than the
+        (type, [data, ...]) tuples returned by other methods, because only one
+        response is represented at a time.
+
+        Example:
+
+        with imap.idle(duration=29 * 60) as idler:
+            for typ, datum in idler:
+                print(typ, datum)
```

Comment:
Can `datum` be a tuple, like the data returned by other commands? Please clarify in docs and docstring either way.